src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -371,7 +367,7 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         private static extern IntPtr /* IUnknown* */ GetComInterfaceForObjectNative(object o, Type t, bool onlyInContext, bool fEnalbeCustomizedQueryInterface);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetObjectForIUnknown(IntPtr /* IUnknown* */ pUnk);
+        public static extern object? GetObjectForIUnknown(IntPtr /* IUnknown* */ pUnk);
```

Comment:
As best as I could tell; the native code supports returning `null`. But, I'm unsure if you can actually give a valid `IntPtr` that would result in that ever happening... Same applies to most of the other functions with a nullable return below.

Reply:
> the native code supports returning null

Where do you see that? I do not think that this can ever return null.

Reply:
I believe it's just a potential case, but `GetObjectRefFromComIP` will definitely assign `pObjOut` to `NULL` if `*ppUnk == NULL` (which shouldn't happen from our managed entry points). Otherwise, it looks like it tries to always find or create a wrapper for the given interface.

I also believe it should fail if you pass in a non-null `IntPtr` which is not an `IUnknown`; but I didn't dig that deeply into it.

I called it out here to ensure I got validation from the people who have more knowledge of CCW/RCW before the PR got merged; and I'll change these to be non-nullable.